src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToAsyncMethods.Analyzer.cs

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    /// <summary>
+    /// CA2016: Forward CancellationToken to async methods.
+    /// 
+    /// Conditions for positive cases:
+    ///     - The containing method signature receives a ct parameter. It can be a method, a nested method, an action or a func.
+    ///     - The invocation method is not receiving a ct argument, and...
+    ///     - The invocation method either:
+    ///         - Has no overloads but its current signature receives an optional ct=default, currently implicit, or...
+    ///         - Has a method overload with the exact same arguments in the same order, plus one ct parameter at the end.
+    ///         
+    /// Conditions for negative cases:
+    ///     - The containing method signature does not receive a ct parameter.
+    ///     - The invocation method signature receives a ct and one is already being explicitly passed, or...
+    ///     - The invocation method does not have an overload with the exact same arguments that also receives a ct, or...
+    ///     - The invocation method only has overloads that receive more than one ct.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
```

Comment:
remove this. We cannot instantiate an abstract class

Reply:
Good catch. I had to create the CSharp and the Basic version of this in my last commit, forgot to remove this.